ada/testsuite/ada/nameres.adb

```diff
@@ -398,6 +401,23 @@ procedure Nameres is
 
          Obj : aliased J.JSON_Value;
 
+         procedure Put_XFAIL_Message is
+            N : constant Ada_Node := Next_Sibling (Node);
+         begin
+            if not Is_Null (N) and then Kind (N) = Ada_Pragma_Node then
+               if Child (N, 1).Text = "XFAIL_Nameres" then
+                  declare
+                     Arg : constant Expr :=
+                       P_Assoc_Expr
+                        (As_Base_Assoc (Child (F_Args (N.As_Pragma_Node), 1)));
+                     Txt : constant String := Text (Arg);
+                  begin
+                     Put_Line ("XFAIL: " & Txt (2 .. Txt'Last - 1));
```

Comment:
I suggest to replace slicing with `Arg.As_String_Literal.P_Denoted_Value`. Also, what bout adding validation code for invalid uses of this new pragma?

Reply:
SGTM about `P_Denoted_Value`, forgot about it. Also wrt validation code, added some, but I don't think litering the code with validation for what is basically an internal pragma is very useful, especially since doing this in Ada is tedious and error prone in itself, so keeping it to a bare minimum.

Reply:
I somewhat agree about validation code cluttering. In the end, Romain and you will be the one dealing with potential crashes, so your call. :-)